packages/desktop-client/src/components/table.tsx

```diff
@@ -1229,19 +1228,15 @@ export function useTableNavigator<T extends TableItem>(
   const containerRef = useRef<HTMLDivElement>();
 
   // See `onBlur` for why we need this
-  const store = useStore();
-  const modalStackLength = useRef(0);
+  const modalState = useModalState();
+  const modalStackLength = useRef(modalState.modalStack.length);
```

Comment:
If we're not using a `useEffect` couldn't this just be a normal variable instead of a ref?

Reply:
It's still needed since we only want the initial modal stack length and not get the updated length on re-renders